contributed_definitions/nyaml/NXsample_component_set.yaml

```diff
@@ -0,0 +1,37 @@
+category: base
+doc: |
+  Set of sample components and their configuration. The idea here is to store all descriptors that 
+  cannot be stored on the level of the individual sample component.
+symbols:
+  doc: |
+    DOC MISSING.
+  n_comp: |
+    number of components
+type: group
+NXsample_component_set(NXobject):
+  layer(NX_CHAR):
+    doc: |
+      Number of layers of the sample (e.g. bulk, monolayer, pentalayer, etc.)
+  concentration(NX_FLOAT):
+    unit: NX_MASS_DENSITY
+    doc: |
+      Concentration of each component
+    dimensions:
+      rank: 1
+      dim: [[1, n_comp]]
+  volume_fraction(NX_FLOAT):
```

Comment:
Keep it there but I can also imaging that this is a descriptor below an NXroi with some depends_on to a set of substances which would explore the complete expressiveness of set theory arbitratily cut out ROIs